src/components/manage/Blocks/Teaser/Data.jsx

```diff
@@ -27,25 +28,14 @@ const TeaserData = (props) => {
     });
   };
 
-  const teaserDataAdapter = ({ block, data, id, onChangeBlock, value }) => {
-    let dataSaved = {
-      ...data,
-      [id]: value,
-    };
-    if (id === 'href' && !isEmpty(value) && !data.title && !data.description) {
-      dataSaved = {
-        ...dataSaved,
-        title: value[0].Title,
-        description: value[0].Description,
-        head_title: value[0].head_title,
-      };
-    }
-    onChangeBlock(block, dataSaved);
-  };
-
   const isReseteable =
     isEmpty(data.href) && !data.title && !data.description && !data.head_title;
 
+  const dataAdapter = config.getComponent({
```

Comment:
I don't know if I'm a fan of this style. I'd be more in favour of an entry in the teaser blockConfig. We already use this pattern, for exampe [`blockHasValue`](https://github.com/plone/volto/blob/e7cf0b141263b099091d64d1ba98751137c839e3/src/config/Blocks.jsx#L225-L230) or [`tocEntry`](https://github.com/plone/volto/blob/e7cf0b141263b099091d64d1ba98751137c839e3/packages/volto-slate/src/blocks/Text/index.js#L130-L138), or `restricted` and so on.

I can list several reasons:

- `dataAdapter` is not a component
- do we have "local component registry" ? We can already tweak blocksConfig "on the fly" and pass it down and this is supported by some of our components. Otherwise we'd have to implement another mechanism for the component registry (which in itself may be worth doing anyway).
- this customization point (being able to override the data adapter) is hidden in the code, rather then being upfront in the block definition
- it's not clear how to use the `dependencies` mechanism to further extend or reuse the Teaser + Data component. If, instead, the code would be `const dataAdapter = config.blocksConfig[data['@type']].dataAdapter`, then that would be make the teaser component reusable in other blocks.

Reply:
I think you are correct. Having it as a block config should be better, I will refactor it.